controllers/device_syncer.go

```diff
@@ -42,83 +42,70 @@ type DeviceSyncer struct {
-	}
-
+func NewDeviceSyncer(client client.Client, opts *options.YurtDeviceControllerOptions, nodepool string) (DeviceSyncer, error) {
 	return DeviceSyncer{
-		syncPeriod: time.Duration(periodSecs) * time.Second,
-		deviceCli:  efCli.NewEdgexDeviceClient(coreMetaCliInfo, coreCmdCliInfo, logr),
+		syncPeriod: time.Duration(opts.EdgeSyncPeriod) * time.Second,
+		deviceCli:  efCli.NewEdgexDeviceClient(opts.CoreMetadataAddr, opts.CoreCommandAddr),
 		Client:     client,
-		log:        log,
-		NodePool:   nodePool,
+		NodePool:   nodepool,
+		Namespace:  opts.Namespace,
 	}, nil
 }
 
-// NewDeviceSyncerRunnablel initialize a controller-runtime manager runnable
-func (ds *DeviceSyncer) NewDeviceSyncerRunnablel() ctrlmgr.RunnableFunc {
+// NewDeviceSyncerRunnable initialize a controller-runtime manager runnable
+func (ds *DeviceSyncer) NewDeviceSyncerRunnable() ctrlmgr.RunnableFunc {
 	return func(ctx context.Context) error {
 		ds.Run(ctx.Done())
 		return nil
 	}
 }
 
 func (ds *DeviceSyncer) Run(stop <-chan struct{}) {
-	ds.log.V(1).Info("starting the DeviceSyncer...")
+	klog.V(1).Info("[Device] Starting the syncer...")
 	go func() {
 		for {
 			<-time.After(ds.syncPeriod)
+			klog.V(2).Info("[Device] Start a round of synchronization.")
 			// 1. get device on edge platform and OpenYurt
 			edgeDevices, kubeDevices, err := ds.getAllDevices()
 			if err != nil {
-				ds.log.V(3).Error(err, "fail to list the devices")
+				klog.V(3).ErrorS(err, "fail to list the devices")
 				continue
 			}
 
 			// 2. find the device that need to be synchronized
 			redundantEdgeDevices, redundantKubeDevices, syncedDevices := ds.findDiffDevice(edgeDevices, kubeDevices)
-			ds.log.V(1).Info("The number of devices waiting for synchronization",
+			klog.V(2).Infof("[Device] The number of objects waiting for synchronization { %s:%d, %s:%d, %s:%d }",
 				"Edge device should be added to OpenYurt", len(redundantEdgeDevices),
 				"OpenYurt device that should be deleted", len(redundantKubeDevices),
 				"Devices that should be synchronized", len(syncedDevices))
 
 			// 3. create device on OpenYurt which are exists in edge platform but not in OpenYurt
 			if err := ds.syncEdgeToKube(redundantEdgeDevices); err != nil {
-				ds.log.V(3).Error(err, "fail to create devices on OpenYurt")
+				klog.V(3).ErrorS(err, "fail to create devices on OpenYurt")
 				continue
 			}
 
 			// 4. delete redundant device on OpenYurt
 			if err := ds.deleteDevices(redundantKubeDevices); err != nil {
-				ds.log.V(3).Error(err, "fail to delete redundant devices on OpenYurt")
+				klog.V(3).ErrorS(err, "fail to delete redundant devices on OpenYurt")
 				continue
 			}
 
 			// 5. update device status on OpenYurt
 			if err := ds.updateDevices(syncedDevices); err != nil {
-				ds.log.Error(err, "fail to update devices status")
+				klog.V(3).ErrorS(err, "fail to update devices status")
 				continue
 			}
-
-			ds.log.V(1).Info("One round of Device synchronization is complete")
```

Comment:
This log may be necessary to indicate whether on round of synchronization completes or not.

Reply:
The log has been added.